definitions/features/downstream.rb

```diff
@@ -55,24 +55,33 @@ def rhsm_refresh
   def rh_repos(sat_version)
     sat_version = version(sat_version)
     rh_version_major = execute!('facter operatingsystemmajrelease')
-    sat_version_full = "#{sat_version.major}.#{sat_version.minor}"
+    rh_repos = main_rh_repos(rh_version_major)
 
+    sat_version_full = "#{sat_version.major}.#{sat_version.minor}"
     sat_repo_id = "rhel-#{rh_version_major}-server-satellite-#{sat_version_full}-rpms"
     sat_tools_repo_id = "rhel-#{rh_version_major}-server-satellite-tools-#{sat_version_full}-rpms"
-    # Override to use Beta repositories for 6.4 until GA
+
     if sat_version.to_s == '6.4'
+      # Override to use Beta repositories for 6.4 until GA
       sat_repo_id = "rhel-server-#{rh_version_major}-satellite-6-beta-rpms"
-    end
 
-    rh_repos = ["rhel-#{rh_version_major}-server-rpms",
-                "rhel-server-rhscl-#{rh_version_major}-rpms",
-                "rhel-#{rh_version_major}-server-satellite-maintenance-6-rpms",
-                sat_tools_repo_id,
-                sat_repo_id]
-    rh_repos << 'rhel-7-server-ansible-2-rpms' if sat_version.to_s == '6.4'
+      rh_repos << 'rhel-7-server-ansible-2-rpms'
+    end
+    rh_repos.concat([sat_tools_repo_id, sat_repo_id])
+    rh_repos << 'rhel-7-server-satellite-tools-6.3-puppet4-rpms' if puppet_repo?(sat_version)
```

Comment:
Shouldn't here be `sat_version` instead of hardcoded `6.3`?

Reply:
Ohh.. my mistake. I will update PR.

Reply:
@iNecas @kgaikwad i think no problem if it is 6.3 hardcoded anyhow puppet4 repo is with 6.3 version only.

Reply:
@iNecas, @ntkathole,

We have considered `puppet4` repository here. 
In similar way, do we need to check about how `puppet5` will be available through main repository or in different repository?

Reply:
@kgaikwad puppet5 will be available from satellite 6.4 repo itself (no separate enabling of puppet5 repo) so no need to check.